### PR TITLE
SALTO-3127: Salesforce unknown picklist values Change Validator

### PR DESCRIPTION
**Pull request by tamtamirr**
Added a ChangeValidator that validates all the changes on instances with Picklist values contain valid values (From the field's `valueSet`).

---

This ChangeValidator relies on the allowed values from the `restriction` core annotation to determine the unknown values.

---
_Release Notes_: 
Salesforce Adapter:
- Unknown picklist value changes will be skipped.

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+} from '@salto-io/adapter-api'
+import { collections, values } from '@salto-io/lowerdash'
+import { isInstanceOfCustomObjectChange } from '../custom_object_instances_deploy'
+import { isPicklistField } from '../filters/value_set'
+
+
+const { isDefined } = values
+const { awu } = collections.asynciterable
+
+const createUnknownPicklistValueChangeError = ({ elemID, annotations }: Field, unknownValue: string): ChangeError => ({
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts:32]: cc. @tomermevorach

File: packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  elemID,
+  message: 'Unknown picklist value',
+  detailedMessage: `The field ${elemID.getFullName()} was set with unsupported value ${unknownValue}. Supported values are ${annotations[CORE_ANNOTATIONS.RESTRICTION]?.values}`,
+  severity: 'Error',
```
Comment (tomermevorach) [inline packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts:36]: message:
Unknown picklist value ${unknownValue}

detailed:
Supported values are ${annotations[CORE_ANNOTATIONS.RESTRICTION]?.values}`

Comment (tamtamirr) [reply packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts:36]: v

File: packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        : createUnknownPicklistValueChangeError(field, fieldValue)
+    })
+    .filter(isDefined)
+}
+
+const changeValidator: ChangeValidator = async changes => (
+  awu(changes)
+    .filter(isInstanceChange)
+    .filter(isAdditionOrModificationChange)
+    .filter(isInstanceOfCustomObjectChange)
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/change_validators/unknown_picklist_values.ts:60]: Why only instances of customObject?

**Review by amitgoldarb — APPROVED** [review]
Approving this, please see if this applies to records of standard objects as well and decide if we need to update the CV accordingly

**Pull request merged**